enterprise_access/apps/api_client/lms_client.py

```diff
@@ -6,16 +6,39 @@
 
 import requests
 from django.conf import settings
+from edx_django_utils.cache import TieredCache
 from rest_framework import status
 
 from enterprise_access.apps.api_client.base_oauth import BaseOAuthClient
 from enterprise_access.apps.api_client.exceptions import FetchGroupMembersConflictingParamsException
+from enterprise_access.cache_utils import versioned_cache_key
 from enterprise_access.utils import should_send_email_to_pecu
 
 logger = logging.getLogger(__name__)
 
 
+def all_enterprise_group_members_cache_key(
```

Comment:
nit: not really "all" if user_query is not null.

Reply:
"all_pages" I think fits a bit better then?